lib/process-services-cloud/src/lib/process/process-filters/components/process-filters-cloud.component.spec.ts

```diff
@@ -345,5 +351,62 @@ describe('ProcessFiltersCloudComponent', () => {
             expect(getActiveFilterElement(runningProcessesFilterKey)).toBeNull();
             expect(getActiveFilterElement(completedProcessesFilterKey)).toBeDefined();
         });
+
+        it('should not emit filter key when filter counter is set for first time', () => {
+            component.currentFiltersValues = {};
+            const fakeFilterKey = 'testKey';
+            const fakeFilterValue = 10;
+            const updatedFilterSpy = spyOn(component.updatedFilter, 'emit');
+            component.checkIfFilterValuesHasBeenUpdated(fakeFilterKey, fakeFilterValue);
+            fixture.detectChanges();
+
+            expect(component.currentFiltersValues).not.toEqual({});
+            expect(component.currentFiltersValues[fakeFilterKey]).toBe(fakeFilterValue);
+            expect(updatedFilterSpy).not.toHaveBeenCalled();
+        });
+
+        it('should not emit filter key when filter counter has not changd', () => {
+            component.currentFiltersValues = {};
+            const fakeFilterKey = 'testKey';
+            const fakeFilterValue = 10;
+            const updatedFilterSpy = spyOn(component.updatedFilter, 'emit');
+            component.checkIfFilterValuesHasBeenUpdated(fakeFilterKey, fakeFilterValue);
+            fixture.detectChanges();
+
+            expect(component.currentFiltersValues).not.toEqual({});
+            expect(component.currentFiltersValues[fakeFilterKey]).toBe(fakeFilterValue);
+
+            component.checkIfFilterValuesHasBeenUpdated(fakeFilterKey, fakeFilterValue);
+            expect(component.currentFiltersValues[fakeFilterKey]).toBe(fakeFilterValue);
+            expect(updatedFilterSpy).not.toHaveBeenCalled();
+        });
+
+        it('should emit filter key when filter counter is increased', (done) => {
+            component.currentFiltersValues = {};
+            const fakeFilterKey = 'testKey';
+            component.checkIfFilterValuesHasBeenUpdated(fakeFilterKey, 10);
+
+            component.updatedFilter.pipe(first()).subscribe((updatedFilter: string) => {
```

Comment:
as on the other part instead of dealing with subscptions and done function we can use firstValueOf and reduce all to a Promise (ti's a unit test in the end) so we can deal with it with async/await